### ISSUE TITLE
Can't read compressed entries larger than 65k

### ISSUE THREAD

**Issue by evanmiller** [nmoinvaz/minizip#227]
I'm trying to figure out why minizip refuses to decompress compressed entries larger than 65k.

First I see this:

```
    if (len > UINT16_MAX) // Zlib limitation
        return MZ_PARAM_ERROR;
```

https://github.com/nmoinvaz/minizip/blob/master/src/mz_zip.c#L1317-L1318

And also this:

```
    zlib->zstream.next_out = (uint8_t*)buf;
    zlib->zstream.avail_out = (uint16_t)size;
```

https://github.com/nmoinvaz/minizip/blob/master/src/mz_strm_zlib.c#L132

Does anyone really have a 16-bit limitation on Zlib? If I change those to the 32-bit equivalents, my ZIP files unpack just fine. The rest of the definitions are 32-bit, and this library used to work fine for me, so I'm confused how the 16-bit limitations got in here.

Thanks for investigating.

**Comment by nmoinvaz** [comment]
I agree with doing a check to see what the size is. If you make a pull request I can merge it.

**Comment by nmoinvaz** [comment]
I have merged your pull request but also taken your approach and changed it a bit. Thank you.
